### PR TITLE
Upgrade to node 18

### PR DESCRIPTION
**Pull request by Will-Howard**
I'm in the process of migrating Rules to Actions, and this [this library](https://www.npmjs.com/package/auth0-actions/v/0.1.1) that has some useful types requires node 18.

I have tested `yarn cli [rules|db|login] diff` and `yarn cli [rules|db|login] deploy` on the local tenant and it worked as expected.

### CODE REVIEW

File: scripts/rules/src/add-default-roles.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ async function addDefaultRoles(
     const err = error as Error
     callback(
       new Error(
-        `Failed to set default role: ${err.message || JSON.stringify(error)}`
+        `Failed to set default role: ${err.message || JSON.stringify(err)}`
```
Comment (Will-Howard) [inline scripts/rules/src/add-default-roles.ts:56]: Unrelated

**Review by oetherington — APPROVED** [review]
LGTM - there's a github action that probably needs updating as well though
